src/metabase/driver/postgres.clj

```diff
@@ -531,13 +538,14 @@
       (assoc :sslrootcert (secret/value->file! ssl-root-cert :postgres))
 
       (has-value? ssl-client-key)
-      (assoc :sslkey (secret/value->file! ssl-client-key :postgres))
+      (assoc :sslkey (secret/value->file! ssl-client-key :postgres (when (pkcs-12-key-value? ssl-key-value) ".p12")))
 
       (has-value? ssl-client-cert)
       (assoc :sslcert (secret/value->file! ssl-client-cert :postgres))
 
-      (has-value? ssl-key-pw)
-      (assoc :sslpassword (secret/value->string ssl-key-pw))
+      ;; Pass an empty string as password if none is provided; otherwise the driver will prompt for one
+      true
+      (assoc :sslpassword (or (secret/value->string ssl-key-pw) ""))
```

Comment:
Have you considered setting `:sslpasswordcallback` instead of setting the password to `""`? Might that let us emit better error messages?

Reply:
I noticed that in the docs but couldn't find lots of details in it. I'm also not totally sure what value it adds...I think it would be useful if we had some way to interactively prompt for a password when the jdbc driver requests it, but since we don't, I think defaulting to an empty string is OK. (I might not be understanding it fully though)